File: Assigment2/MusicStudioRentalApp/app/src/main/java/com/example/musicstudiorental/utils/ParcelableUtil.java
package com.example.musicstudiorental.utils;

import android.os.Parcel;
import android.os.Parcelable;

public class ParcelableUtil {

    public static <T extends Parcelable> T fromParcel(Parcel in, Parcelable.Creator<T> creator) {
        return creator.createFromParcel(in);
    }

    public static <T extends Parcelable> void writeToParcel(T item, Parcel out) {
        item.writeToParcel(out, 0);
    }
}

File: Assigment2/app/src/main/java/com/example/musi/utils/ParcelableUtil.kt
package com.example.musi.utils

import android.os.Parcel
import android.os.Parcelable

object ParcelableUtil {
    fun <T : Parcelable?> fromParcel(`in`: Parcel?, creator: Parcelable.Creator<T>): T {
        return creator.createFromParcel(`in`)
    }

    fun <T : Parcelable?> writeToParcel(item: T, out: Parcel) {
        item!!.writeToParcel(out, 0)
    }
}